File: frontend/src/types/inventory.ts
export interface InventoryItem {
  id: number;
  event_id: number;
  section: string;
  row: string | null;
  seat_numbers: string | null;
  quantity: number;
  cost_per_ticket: number;
  total_cost: number;
  purchase_date: string | null;
  target_sell_min: number | null;
  target_sell_max: number | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
  // Market data
  current_market_price: number | null;
  expected_revenue: number | null;
  expected_profit: number | null;
  comparable_listings_count: number;
  min_market_price: number | null;
  max_market_price: number | null;
  avg_market_price: number | null;
}

File: backend/app/schemas/inventory.py
from datetime import datetime, date
from decimal import Decimal
from pydantic import BaseModel, computed_field


class InventoryBase(BaseModel):
    event_id: int
    section: str
    row: str | None = None
    seat_numbers: str | None = None
    quantity: int = 1
    cost_per_ticket: Decimal
    total_cost: Decimal
    purchase_date: date | None = None
    target_sell_min: Decimal | None = None
    target_sell_max: Decimal | None = None
    notes: str | None = None


class InventoryCreate(InventoryBase):
    pass


class InventoryUpdate(BaseModel):
    section: str | None = None
    row: str | None = None
    seat_numbers: str | None = None
    quantity: int | None = None
    cost_per_ticket: Decimal | None = None
    total_cost: Decimal | None = None
    purchase_date: date | None = None
    target_sell_min: Decimal | None = None
    target_sell_max: Decimal | None = None
    notes: str | None = None


class InventoryResponse(InventoryBase):
    id: int
    created_at: datetime
    updated_at: datetime

    # Computed fields for UI display
    current_market_price: Decimal | None = None
    expected_revenue: Decimal | None = None
    expected_profit: Decimal | None = None

    class Config:
        from_attributes = True


class InventoryWithMarketData(InventoryResponse):
    """Inventory item with current market pricing data attached"""
    comparable_listings_count: int = 0
    min_market_price: Decimal | None = None
    max_market_price: Decimal | None = None
    avg_market_price: Decimal | None = None
